Simplified/NYPLBarcode.swift

```diff
@@ -0,0 +1,56 @@
+import Foundation
+import ZXingObjC
+
+fileprivate let barcodeHeight: CGFloat = 100
+
+@objc enum NYPLBarcodeType: Int {
+  case codabar
+  case code39
+  case qrCode
+  case code128
+}
+
+fileprivate func ZXBarcodeFormatFor(_ NYPLBarcodeType:NYPLBarcodeType) -> ZXBarcodeFormat {
+  switch NYPLBarcodeType {
+  case .codabar:
+    return kBarcodeFormatCodabar
+  case .code39:
+    return kBarcodeFormatCode39
+  case .qrCode:
+    return kBarcodeFormatQRCode
+  case .code128:
+    return kBarcodeFormatCode128
+  }
+}
+
+/// Manage creation of barcode images that are scannable by physical scanners.
+/// Keep any third party dependency abstracted out of the main app.
+final class NYPLBarcode: NSObject {
+
+  class func image(string: String, size: CGSize, type: NYPLBarcodeType) -> UIImage?
+  {
+    let writer = ZXMultiFormatWriter.writer() as? ZXWriter
+    do {
```

Comment:
Do we really want to continue here if `writer` is nil? It seems like this can result in a nil `result` being passed to `ZXImage.init` a few lines down.